### ISSUE TITLE
Unreliable behavior of delete_password() across operating systems

### ISSUE THREAD

**Issue by moritzheiber** [hwchen/keyring-rs#39]
After [your comment](https://github.com/hwchen/keyring-rs/pull/38#issuecomment-602881762) I've created a short POC which (reproducably) succeeds on Linux, but fails on macOS and Windows:

```rust
extern crate keyring;

use keyring::{Keyring, KeyringError};

fn main() -> Result<(), KeyringError> {
    let service = "ce06501c7f791b4d653effb50f234cd6fa451c734054acb9936c14e965abf37d";
    let username = "username";

    let keyring = Keyring::new(service, username);
    match keyring.delete_password() {
        Err(KeyringError::NoPasswordFound) => Ok(()),
        Err(e) => Err(e),
        _ => Ok(()),
    }
}
```
On Linux this just reaches the `Ok(())` of the `NoPasswordFound` arm, on Windows it produces a `WindowsVaultError`, on macOS it returns a general `KeyringError` with the message `Mac Os Keychain Error: The specified item could not be found in the keychain.`.

I would love for the function to return a predictable result, ideally :slightly_smiling_face: 

(And have a test case for it!)

**Comment by hwchen** [comment]
Thanks for this example! I'll take a look at it and have better/consistent error handling. I think that this will also affect the errors for `.get_password`.

**Comment by hwchen** [comment]
I will be handling this case for sure; but I may also decide to do some more cleaning of error handling. See #40

**Comment by moritzheiber** [comment]
Here are the only available error states for Secret Service (and therefore Linux): https://specifications.freedesktop.org/secret-service/latest/ch15.html

**Comment by adobeDan** [comment]
@moritzheiber Just wanted you to know that this is being addressed by #70, which introduces consistent error codes across platforms.

**Comment by brotskydotcom** [comment]
closed with merge of #70.
